python-package/xgboost/core.py

```diff
@@ -1056,7 +1063,12 @@ def predict(self, data, output_margin=False, ntree_limit=0, pred_leaf=False,
         nrow = data.num_row()
         if preds.size != nrow and preds.size % nrow == 0:
             ncol = int(preds.size / nrow)
-            preds = preds.reshape(nrow, ncol)
+            if interaction_contribs:
+                s = int(np.sqrt(ncol))
+                assert s*s == ncol, "XGBoost returned a non-square interaction matrix!"
+                preds = preds.reshape(nrow, int(np.sqrt(ncol)), int(np.sqrt(ncol)))
```

Comment:
This doesn't work for multiclass.

Reply:
good point. should the regular labels also break things out by class as a dimension? It would be a breaking change, but might make more sense

Reply:
If the order of indexing would be well documented, maybe with some brief examples, then breaking things by class is not really necessary in python, as reshaping is quite simple and common operation for numpy arrays. The "discoverability" would be better though if things come pre-shaped... So I don't have much of an opinion towards either way here.

Reply:
I went ahead and pre-shaped them for both interactions and regular contributions...since digging into the docs is less pleasant that having it be obvious.